[DOCS] OVC/convert_model Documentation (#19555)
openvinotoolkit/openvino: docs/OV_Converter_UG/prepare_model/convert_model/MO_OVC_transition.md

             from openvino.tools.mo import convert_model
             ov_model = convert_model(model, scale_value=[255., 255., 255.])

* scale parameter:

   .. tab-set::

       .. tab-item:: ov.convert_model()
          :sync: py

          .. code-block:: py
             :force:

             from openvino import convert_model
             ov_model = convert_model(model)

             from openvino.preprocess import PrePostProcessor
             prep = PrePostProcessor(ov_model)
             prep.input(input_name).preprocess().scale(255.)
             ov_model = prep.build()

       .. tab-item:: mo.convert_model()
          :sync: py

          .. code-block:: py
             :force:

             from openvino.tools.mo import convert_model
             ov_model = convert_model(model, scale=255.)

* reverse_input_channels parameter:

   .. tab-set::

       .. tab-item:: ov.convert_model()
          :sync: py

          .. code-block:: py
             :force:

             from openvino import convert_model
             ov_model = convert_model(model)

             from openvino.preprocess import PrePostProcessor
             prep = PrePostProcessor(ov_model)

>              prep.input(input_name).preprocess().reverse_channels()

Reviewer: ```suggestion
             prep.input(input_name).tensor().set_layout(Layout(layout_string))
             prep.input(input_name).preprocess().reverse_channels()
```

# Transition from Legacy Conversion API {#openvino_docs_OV_Converter_UG_prepare_model_convert_model_MO_OVC_transition}

@sphinxdirective

.. meta::
   :description: Transition guide from MO / mo.convert_model() to OVC / ov.convert_model().

.. toctree::
   :maxdepth: 1
   :hidden:

   openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide

> In 2023.1 OpenVino release a new OVC (OpenVINO Model Converter) tool was introduced with the corresponding Python API: ``openvino.convert_model`` method. ``ovc`` and ``openvino.convert_model`` represent

Reviewer 2: ```suggestion
In 2023.1 OpenVINO release a new OVC (OpenVINO Model Converter) tool has been introduced with the corresponding Python API: ``openvino.convert_model`` method. ``ovc`` and ``openvino.convert_model`` represent
```

> a lightweight alternative of ``mo`` and ``openvino.tools.mo.convert_model`` which are considered legacy API now. In this article, all the differences between ``mo`` and ``ovc`` are summarized and transition guide from the legacy API to the new API is provided.

Reviewer 2: ```suggestion
a lightweight alternative of ``mo`` and ``openvino.tools.mo.convert_model`` which are considered legacy API now. In this article, all the differences between ``mo`` and ``ovc`` are summarized and the transition guide from the legacy API to the new API is provided.
```

> Parameters comparison

Reviewer 2: ```suggestion
Parameters Comparison
```

#####################

The comparison of parameters between ov.convert_model() / OVC and mo.convert_model() / MO.

.. list-table::
   :widths: 20 25 55
   :header-rows: 1

   * - mo.convert_model() / MO
     - ov.convert_model() / OVC
     - Differences description
   * - input_model
     - input_model

>      - Along with model object or path to input model ov.convert_model() accepts list of model parts, for example path to Tensorflow weights plus path to Tensorflow checkpoint. OVC tool accepts unnamed input model.

Reviewer 2: ```suggestion
     - Along with model object or path to input model ov.convert_model() accepts list of model parts, for example, the path to TensorFlow weights plus the path to TensorFlow checkpoint. OVC tool accepts an unnamed input model.
```

   * - enable_ssd_gluoncv
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - input_proto
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - caffe_parser_path
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - k
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - disable_omitting_optional
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - enable_flattening_nested_params
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - counts
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - remove_output_softmax
     - N/A
     - Not available in ov.convert_model() / OVC.
   * - remove_memory
     - N/A
     - Not available in ov.convert_model() / OVC.

Transition from legacy API to new API
############################################################################

mo.convert_model() provides a wide range of preprocessing parameters. Most of these parameters have analogs in OVC or can be replaced with functionality from ``ov.PrePostProcessor`` class.
Here is the guide to transition from legacy model preprocessing to new API preprocessing.

> ``input_shape``:

Reviewer 2: this might seem unnecessary but is done for the sake of the right hand menu http://openvino-doc.iotg.sclab.intel.com/mo_docs/openvino_docs_OV_Converter_UG_prepare_model_convert_model_MO_OVC_transition.html#input-shape

            - .. code-block:: sh
                 :force:

                 ovc MODEL_NAME --input [1,3,100,100],[1] --output_model OUTPUT_MODEL

``batch``:
##########

.. tab-set::

    .. tab-item:: Python
       :sync: py

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: py
                 :force:

                 from openvino.tools import mo

                 ov_model = mo.convert_model(model, batch=2)

            - .. code-block:: py
                 :force:

                 import openvino as ov

                 ov_model = ov.convert_model(model)
                 input_shape = ov_model.inputs[0].partial_shape
                 input_shape[0] = 2 # batch size
                 ov_model.reshape(input_shape)

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --batch 2 --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> ``mean_values``:

Reviewer 2: ```suggestion
``mean_values``
```

                 ov_model = mo.convert_model(model, mean_values=[0.5, 0.5, 0.5])

            - .. code-block:: py
                 :force:

                 import openvino as ov

                 ov_model = ov.convert_model(model)

                 prep = ov.preprocess.PrePostProcessor(ov_model)
                 prep.input(input_name).tensor().set_layout(ov.Layout("NHWC"))
                 prep.input(input_name).preprocess().mean([0.5, 0.5, 0.5])
                 ov_model = prep.build()

              There is currently no heuristic for automatic detection of the channel to which mean, scale or reverse channels should be applied. ``Layout`` needs to be explicitly specified with "C" channel. For example "NHWC", "NCHW", "?C??". See also :doc:`Layout API overview <openvino_docs_OV_UG_Layout_Overview>`.

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --mean_values [0.5,0.5,0.5] --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> ``scale_values``:

Reviewer 2: ```suggestion
``scale_values``
```

                 ov_model = mo.convert_model(model, reverse_input_channels=True)

            - .. code-block:: py
                 :force:

                 import openvino as ov

                 ov_model = ov.convert_model(model)

                 prep = ov.preprocess.PrePostProcessor(ov_model)
                 prep.input(input_name).tensor().set_layout(ov.Layout("NHWC"))
                 prep.input(input_name).preprocess().reverse_channels()
                 ov_model = prep.build()

              There is currently no heuristic for automatic detection of the channel to which mean, scale or reverse channels should be applied. ``Layout`` needs to be explicitly specified with "C" channel. For example "NHWC", "NCHW", "?C??". See also :doc:`Layout API overview <openvino_docs_OV_UG_Layout_Overview>`.

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --reverse_input_channels --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> ``source_layout``:

Reviewer 2: ```suggestion
``source_layout``
```

``target_layout``:
##################

.. tab-set::

    .. tab-item:: Python
       :sync: py

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: py
                 :force:

                 import openvino as ov
                 from openvino.tools import mo

                 ov_model = mo.convert_model(model, target_layout={input_name: ov.Layout("NHWC")})

            - .. code-block:: py
                 :force:

                 import openvino as ov

                 ov_model = ov.convert_model(model)

                 prep = ov.preprocess.PrePostProcessor(ov_model)
                 prep.input(input_name).tensor().set_layout(ov.Layout("NHWC"))
                 ov_model = prep.build()

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --target_layout input_name(NHWC) --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> ``layout``:

Reviewer 2: ```suggestion
``layout``
```

###########

.. tab-set::

    .. tab-item:: Python
       :sync: py

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: py
                 :force:

                 from openvino.tools import mo

                 ov_model = mo.convert_model(model, layout={input_name: mo.LayoutMap("NCHW", "NHWC")})

            - .. code-block:: py
                 :force:

                 import openvino as ov

                 ov_model = ov.convert_model(model)

                 prep = ov.preprocess.PrePostProcessor(ov_model)
                 prep.input(input_name).model().set_layout(ov.Layout("NCHW"))
                 prep.input(input_name).tensor().set_layout(ov.Layout("NHWC"))
                 ov_model = prep.build()

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --layout "input_name(NCHW->NHWC)" --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> ``transform``:

Reviewer 2: ```suggestion
``transform``
```

            - .. code-block:: py
                 :force:

                 import openvino as ov
                 from openvino._offline_transformations import apply_low_latency_transformation, apply_pruning_transformation, apply_make_stateful_transformation

                 ov_model = ov.convert_model(model)
                 apply_low_latency_transformation(model, use_const_initializer=False)
                 apply_pruning_transformation(model)
                 apply_make_stateful_transformation(model, param_res_names={'input_name': 'output_name'})

    .. tab-item:: CLI
       :sync: cli

       .. list-table::
          :header-rows: 1

          * - Legacy API
            - New API
          * - .. code-block:: sh
                 :force:

                 mo --input_model MODEL_NAME --transform LowLatency2[use_const_initializer=False],Pruning,MakeStateful[param_res_names={'input_name':'output_name'}] --output_dir OUTPUT_DIR

            - Not available in OVC tool. Please check Python API.

> Supported frameworks in MO vs OVC

Reviewer 2: ```suggestion
Supported Frameworks in MO vs OVC
```

#################################

ov.convert_model() and OVC tool support conversion from PyTorch, TF, TF Lite, ONNX, PaddlePaddle.

> Following frameworks are supported only in MO and mo.convert_model(): Caffe, MxNet, Kaldi.

Reviewer 2: ```suggestion
The following frameworks are supported only in MO and mo.convert_model(): Caffe, MxNet, Kaldi.
```
